lib/net.js

```diff
@@ -1404,6 +1404,12 @@ Server.prototype.listen = function() {
   var options = normalized[0];
   var cb = normalized[1];
 
+  if (this._handle) {
+    var er = new Error('listen method has been called more than once.');
```

Comment:
The other thing I would recommend here is using the new `internal/errors` when introducing a new error.